code/modules/integrated_electronics/subtypes/data_transfer.dm

```diff
@@ -145,6 +145,55 @@
 	w_class = WEIGHT_CLASS_SMALL
 	number_of_pins = 16
 
+/obj/item/integrated_circuit/transfer/pulsemultiplexer
+	name = "two pulse multiplexer"
+	desc = "Pulse in pins to choose the pin value to be sent."
+	extended_desc = "The input pulses are used to select which of the input pins has its data moved to the output."
+	complexity = 2
+	icon_state = "dmux2"
+	inputs = list()
+	outputs = list("output" = IC_PINTYPE_ANY)
+	activators = list("on selected" = IC_PINTYPE_PULSE_OUT)
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+	power_draw_per_use = 4
+	var/number_of_pins = 2
+
+/obj/item/integrated_circuit/transfer/pulsemultiplexer/Initialize()
+	for(var/i = 1 to number_of_pins)
+		inputs["input [i]"] = IC_PINTYPE_ANY
+	for(var/i = 1 to number_of_pins)
+		activators["input [i]"] = IC_PINTYPE_PULSE_IN
+	complexity = number_of_pins
+
+	. = ..()
+	desc += " It has [number_of_pins] input pins and [number_of_pins] output pins."
```

Comment:
hey did you mean to say pulse pins instead of output pins?

Reply:
No, I meant to say pulse pins instead of input pins. Nice catch there, though.